src/image-gif.c

```diff
@@ -0,0 +1,707 @@
+/*
+ * Twin - A Tiny Window System
+ * Copyright (c) 2024 National Cheng Kung University, Taiwan
+ * All rights reserved.
+ *
+ *
+ * This file integrates the GIF decoder originally developed by Marcel Rodrigues
+ * as part of the gifdec project. You can find the original gifdec project at:
+ * https://github.com/lecram/gifdec
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "twin.h"
+#include "twin_private.h"
+
+typedef struct _twin_gif_palette {
+    int size;
+    uint8_t colors[0x100 * 3];
+} twin_gif_palette_t;
+
+typedef struct _twin_gif_gce {
+    uint16_t delay;
+    uint8_t tindex;
+    uint8_t disposal;
+    int input;
+    int transparency;
+} twin_gif_gce_t;
+
+typedef struct _twin_gif {
+    int fd;
+    off_t anim_start;
+    twin_coord_t width, height;
+    twin_coord_t depth;
+    twin_count_t loop_count;
+    twin_gif_gce_t gce;
+    twin_gif_palette_t *palette;
+    twin_gif_palette_t lct, gct;
+    void (*plain_text)(struct _twin_gif *gif,
```

Comment:
Is the callback `plain_text` necessary?

Reply:
We don't handle the plain text section now, so it could be removed. Perhaps the callback `comment` and `application` for corresponding sections could be removed as well?